### PR TITLE
Add IDTokenSource Interface

### PR DESCRIPTION
**Pull request by emmyzhou-db**
## What changes are proposed in this pull request?
This PR adds the foundational components for OIDC support in the Databricks Java SDK:

**IDToken Class:** Encapsulates ID tokens from identity providers for OAuth flows.
**IDTokenSource** Interface: Defines a standard for retrieving ID tokens.

## How is this tested?
- Unit tests added for the IDToken class.
- IDTokenSource is not tested, as no implementation exists yet.

NO_CHANGELOG=true

### CODE REVIEW

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/IDToken.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  /**
+   * Constructs an IDToken with a value.
+   *
+   * @param value The ID Token string.
+   */
+  public IDToken(String value) {
+    this.value = value;
+  }
+
+  public String getValue() {
```
Comment (renaudhartert-db) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/IDToken.java:20]: Because I feel like I have to say something, could you please add a small comment on this method?

**Review by renaudhartert-db — APPROVED** [review]
LGTM modulo open comment

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/IDToken.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  private final String value;
+
+  /**
+   * Constructs an IDToken with a value.
+   *
+   * @param value The ID Token string.
+   */
+  public IDToken(String value) {
+    this.value = value;
+  }
```
Comment (parthban-db) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/IDToken.java:18]: Add a null check in the constructor.
```suggestion
  public IDToken(String value) {
    if (value == null || value.isEmpty()) {
        throw new IllegalArgumentException("ID Token value cannot be null or empty");
    }
    this.value = value;
  }
```

File: databricks-sdk-java/src/test/java/com/databricks/sdk/core/oauth/IDTokenTest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public class IDTokenTest {
+
+  private static final String accessToken = "testIdToken";
+
+  @Test
+  void createIDToken() {
+    IDToken idToken = new IDToken(accessToken);
+    assertEquals(accessToken, idToken.getValue());
+  }
```
Comment (parthban-db) [inline databricks-sdk-java/src/test/java/com/databricks/sdk/core/oauth/IDTokenTest.java:15]: Add test for the null value.
